File: hooks/scripts/console-log-warning.js
#!/usr/bin/env node
/**
 * Console.log Warning Hook
 *
 * Runs on PostToolUse for Edit tool.
 * Warns if console.log statements were added.
 */

const fs = require('fs');

function main() {
  let input = '';

  try {
    input = fs.readFileSync(0, 'utf-8');
  } catch {
    process.exit(0);
  }

  // Only check JS/TS files
  let data;
  try {
    data = JSON.parse(input);
  } catch {
    process.exit(0);
  }

  const filePath = data?.tool_input?.file_path || '';
  if (!/\.(js|jsx|ts|tsx|mjs|cjs)$/.test(filePath)) {
    process.exit(0);
  }

  const newString = data?.tool_input?.new_string || '';
  if (newString.includes('console.log')) {
    const message = '⚠️ console.log detected - remember to remove before commit';
    console.log(JSON.stringify({
      systemMessage: message,
      hookSpecificOutput: {
        hookEventName: "PostToolUse",
        additionalContext: message
      }
    }));
  }

  process.exit(0);
}

main();

File: hooks/scripts/console_log_warning.py
#!/usr/bin/env python3
"""PostToolUse hook: warn when console.log statements are added to JS/TS files."""

from __future__ import annotations

import json
import re
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent))
from _util import read_hook_stdin


def check(data: dict) -> str | None:
    """Check if the edit introduces console.log statements.

    Accepts the parsed hook stdin data dict. Returns the warning
    message string if console.log is detected, or None.
    """
    if not data:
        return None

    tool_input = data.get("tool_input", {})
    file_path = tool_input.get("file_path", "") if isinstance(tool_input, dict) else ""

    if not re.search(r"\.(js|jsx|ts|tsx|mjs|cjs)$", file_path):
        return None

    new_string = tool_input.get("new_string", "") if isinstance(tool_input, dict) else ""
    content = tool_input.get("content", "") if isinstance(tool_input, dict) else ""
    text = new_string or content
    if "console.log" in text:
        return "console.log detected - remember to remove before commit"

    return None


def main() -> int:
    data = read_hook_stdin()
    if not data:
        return 0

    result = check(data)
    if result:
        print(
            json.dumps(
                {
                    "systemMessage": result,
                    "hookSpecificOutput": {
                        "hookEventName": "PostToolUse",
                        "additionalContext": result,
                    },
                }
            )
        )

    return 0


if __name__ == "__main__":
    sys.exit(main())
